enhancements/20230316-baremetal-container-mcm-support.md

```diff
@@ -0,0 +1,151 @@
+    ![harvester-service.png](./20230316-baremetal-container-mcm-support/harvester-service.png)
+   - In the `Cluster Tools` page, we should disable operations that could affect Harvester, such as edit/delete monitor, longhorn, and logging.
+    ![cluster-tools.png](./20230316-baremetal-container-mcm-support/cluster-tools.png)
+   - If the `local` cluster is a harvester cluster, we need to hide resources prefixed with `harvesterhci.io.*`, and `kubevirt.io.*`
+    ![harvester-crd.png](./20230316-baremetal-container-mcm-support/harvester-crd.png)
+
+### Implementation Overview
+
+1. Turn on the `multi-cluster-management` feature in the embedded Rancher server config.
+3. Add a new setting `rancher-manager-support` to the Harvester setting page that will allow configuration in both installation and post-installation stage.
+4. Customize the Rancher UI to show/hide the explorer UI and the multi-cluster management UI based on the `rancher-manager-support` setting. 
+    - Showing both the local Harvester cluster and imported Harvester clusters on the explorer and continues delivery clusters.
+    - Hide overlapped and undesired configurations(e.g., disable Harvester and MCM feature) in both Rancher's global setting page and the local Harvester cluster setting page.
+    - Replace the `Home` with `Harvester Dashboard` by default going to the local Harvester Dashboard to provide consistent user experience of single and multi-cluster views.
+
+
+### Test plan
+
+- Users should be able to enable/disable the container workloads management feature via the setting page, and the explorer UI will be shown in the hamburger menu.
+- Users should be able to enable/disable multi-cluster view through the `rancher-manager-support` setting, either before or after installation, and the related UI will be shown in the hamburger menu.
+- Users should be able to visit both the local Harvester cluster and imported Harvester/k8s clusters explorer UI.
+- Users should be able to visit both the local Harvester cluster and imported Harvester clusters virtualization management UI.
+- Users can spin-up guest k8s clusters in the local bare-metal cluster through Harvester node driver.
+- Users can enable authentication and authorization on the bare-metal Harvester cluster, and the [multi-tenancy](https://ranchermanager.docs.rancher.com/v2.6/pages-for-subheaders/authentication-config) feature on the local Harvester cluster should be identical to a seperated upstream Rancher deployment.
+
+### Upgrade strategy
+
+- According to Rancher's [doc](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/feature-flags), the `multi-cluster-management` is used for multi-cluster provisioning and management of Kubernetes clusters. This feature flag can only be set at install time and not changed afterwards. Therefore, we will need to check the upgrade path since this is disabled by default in the Harvester's embedded Rancher.
```

Comment:
```suggestion
- According to Rancher's [documentation](https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/feature-flags), the  feature flag `multi-cluster-management` is used for multi-cluster provisioning and management of Kubernetes clusters. This feature flag can only be configured at the time of installation and can't be changed after installation. Therefore, we will need to check the upgrade path since this is disabled by default in the Harvester's embedded Rancher.
```